### PR TITLE
Python 3: port tests in workers

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

**Review by jdm — APPROVED** [review]

**Review by Hexcles — APPROVED** [review]
LGTM % a nit

File: workers/semantics/encodings/003-1.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
  # -*- coding: utf-8 -*-
+from wptserve.utils import isomorphic_encode
 
 def main(request, response):
-    return "PASS" if request.GET.first('x') == 'å' else "FAIL"
+    return u"PASS" if request.GET.first(b'x') == isomorphic_encode('å') else u"FAIL"
```
[the remark this answers is not in the document]
Comment (ziransun) [reply workers/semantics/encodings/003-1.py:5]: > return u"PASS" if request.GET.first(b'x').decode('utf-8') == u'å' else u"FAIL"
gives Error "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe5 in position 0: unexpected end of data". Any thoughts?

[the remark this answers is not in the document]
Comment (Hexcles) [reply workers/semantics/encodings/003-1.py:5]: What... Could you print out `request.GET.first(b'x')` and see what it is?

[the remark this answers is not in the document]
Comment (Hexcles) [reply workers/semantics/encodings/003-1.py:5]: Actually I think there's a bug in wptserve...

[the remark this answers is not in the document]
Comment (Hexcles) [reply workers/semantics/encodings/003-1.py:5]: Please wait for #24123 to land.

[the remark this answers is not in the document]
Comment (Hexcles) [reply workers/semantics/encodings/003-1.py:5]: That PR has now landed. Could you rebase and try again and see if it works now?

[the remark this answers is not in the document]
Comment (ziransun) [reply workers/semantics/encodings/003-1.py]: It works well :).

**Comment by Hexcles** [comment]
There's an existing flake for `/workers/baseurl/alpha/sharedworker-in-worker.html`; I'm going to admin-merge this.

**Pull request merged**